cmd/query/app/querysvc/query_service.go

```diff
@@ -38,13 +39,14 @@ const (
 type QueryServiceOptions struct {
 	ArchiveSpanReader spanstore.Reader
 	ArchiveSpanWriter spanstore.Writer
+	MetricsReader     metricsstore.Reader
```

Comment:
Query service constructor accepts metric reader as argument. Having it also as a field in these options is redundant and confusing to the callers of the constructor

Reply:
I don't think the query service constructor accepts metric reader as an argument . It only accepts span reader, dependency reader and options.

Reply:
Then it's even worse, if QueryService struct is only designed to serve _trace queries_, there's absolutely no reason to introduce metrics reader into it. Metrics reader is actually a counterpart for QueryService. It's a poor design to couple them in any way.

Notice that QueryService itself does not serve any remote endpoints, it's purely an internal API. The capabilities struct is served via remote endpoint by `app.Server` (which does have access to move qs and mqs), and this is where the logic should be, ideally.